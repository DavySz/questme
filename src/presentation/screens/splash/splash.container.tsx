import { Splash } from "./splash";

export const SplashContainer = (): ReturnType<React.FC> => {
  return <Splash />;
};
