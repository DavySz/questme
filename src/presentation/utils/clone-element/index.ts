export * from "./clone-element-with-props";
