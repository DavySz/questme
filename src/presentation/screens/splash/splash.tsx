import { ScreenTemplate, Text } from "@presentation/components";
import LogoIcon from "@presentation/assets/logo-icon.svg";
import { Content } from "./splash.styles";

export const Splash = (): ReturnType<React.FC> => {
  return (
    <ScreenTemplate.Root variant="tertiary">
      <Content>
        <LogoIcon />
        <Text.Root color="neutral-white" variant="heading-3">
          Questme
        </Text.Root>
      </Content>
    </ScreenTemplate.Root>
  );
};
