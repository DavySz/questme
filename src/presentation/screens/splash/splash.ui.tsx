import { ScreenTemplate, Text } from "@presentation/components";
import LogoIcon from "@presentation/assets/icons/logo-icon.svg";
import { Content } from "./splash.styles";

export const SplashUI = (): ReturnType<React.FC> => {
  return (
    <ScreenTemplate.Root variant="tertiary">
      <Content>
        <LogoIcon />
        <Text.Root color="neutral-white" variant="heading-3">
          Queezy
        </Text.Root>
      </Content>
    </ScreenTemplate.Root>
  );
};
