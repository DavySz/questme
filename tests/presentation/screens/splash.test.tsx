import { SplashContainer } from "@presentation/screens/splash/splash.container";
import { customRender } from "../../utils/custom-render";

describe("Splash", () => {
  test("Should render without errors", () => {
    const tree = customRender(<SplashContainer />);
    expect(tree).toMatchSnapshot();
  });
});
