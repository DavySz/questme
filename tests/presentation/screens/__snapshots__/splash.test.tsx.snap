// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Should render without errors 1`] = `
<View
  style={
    {
      "backgroundColor": "#6A5AE0",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
  testID="bubbles-container"
>
  <View
    left={56}
    style={
      {
        "left": 56,
        "position": "absolute",
        "top": 174,
      }
    }
    top={174}
  >
    <View
      size={48}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "height": 48,
          "opacity": 0.1,
          "width": 48,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={200}
    right={78}
    style={
      {
        "bottom": 200,
        "position": "absolute",
        "right": 78,
      }
    }
  >
    <View
      size={20}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 20,
          "opacity": 0.1,
          "width": 20,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={0}
    left={-121}
    style={
      {
        "left": -121,
        "position": "absolute",
      }
    }
  >
    <View
      size={242}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 121,
          "borderBottomRightRadius": 121,
          "borderTopLeftRadius": 121,
          "borderTopRightRadius": 121,
          "height": 242,
          "opacity": 0.1,
          "width": 242,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={-83.6}
    left={-209}
    style={
      {
        "bottom": -83.6,
        "left": -209,
        "position": "absolute",
      }
    }
  >
    <View
      size={418}
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 209,
          "borderBottomRightRadius": 209,
          "borderColor": "#C9F2E9",
          "borderStyle": "solid",
          "borderTopLeftRadius": 209,
          "borderTopRightRadius": 209,
          "borderWidth": 1,
          "height": 418,
          "opacity": 0.1,
          "width": 418,
        }
      }
      type="outline"
    />
  </View>
  <View
    right={-121}
    style={
      {
        "position": "absolute",
        "right": -121,
      }
    }
    top={0}
  >
    <View
      size={242}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 121,
          "borderBottomRightRadius": 121,
          "borderTopLeftRadius": 121,
          "borderTopRightRadius": 121,
          "height": 242,
          "opacity": 0.1,
          "width": 242,
        }
      }
      type="fill"
    />
  </View>
  <View
    right={-209}
    style={
      {
        "position": "absolute",
        "right": -209,
        "top": -83.6,
      }
    }
    top={-83.6}
  >
    <View
      size={418}
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 209,
          "borderBottomRightRadius": 209,
          "borderColor": "#C9F2E9",
          "borderStyle": "solid",
          "borderTopLeftRadius": 209,
          "borderTopRightRadius": 209,
          "borderWidth": 1,
          "height": 418,
          "opacity": 0.1,
          "width": 418,
        }
      }
      type="outline"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "gap": 16,
        "justifyContent": "center",
      }
    }
    variant="tertiary"
  >
    <SvgMock />
    <Text
      color="neutral-white"
      style={
        {
          "color": "#FFF",
          "fontFamily": "Rubik_500Medium",
          "fontSize": 24,
          "lineHeight": 36,
        }
      }
      variant="heading-3"
    >
      Queezy
    </Text>
  </View>
</View>
`;
